### PR TITLE
Chrisdemars/focus within

### PR DESCRIPTION
**Pull request by chrisdemars**
I added the `:focus-within` snippet that gives focus to any parent element with the pseudo class appended when a child element receives focus like an `input`.

### CODE REVIEW

File: snippets/focus-within.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+### Focus Within
+
+The psuedo class `:focus-within` gives focus to a parent element like a `form` element when `input` elements receive focus.
```
Comment (flxwu) [inline snippets/focus-within.md]: I'd reword this to "... applies styles to a parent element if any child element gets focused. For example, ..."

**Comment by chrisdemars** [comment]
I am submitting a new PR with the change. I think I submitted it? If not please let me know.

File: snippets/focus-within.md (hunk 1 above)
Comment (chrisdemars) [reply snippets/focus-within.md]: I did and now I don't know how to submit a new PR lol.

File: snippets/focus-within.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+```html
+  <form>
+    <label for="given_name">Given Name:</label>
+    <input id="given_name" type="text">
+    <br>
+    <label for="family_name">Family Name:</label>
+    <input id="family_name" type="text">
+  </form>
+
```
Comment (flxwu) [inline snippets/focus-within.md]: unused line

File: snippets/focus-within.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  border: 3px solid #2d98da;
+  color:  #000000;
+  padding: 4px;
+}
+
+form:focus-within {
+  background: #f7b731;
+  color: #000000;
+}
+
```
Comment (flxwu) [inline snippets/focus-within.md]: UL

File: snippets/focus-within.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  </form>
+
+```
+
+#### CSS
+
+```css
+form {
+  border: 3px solid #2d98da;
+  color:  #000000;
```
Comment (flxwu) [inline snippets/focus-within.md]: unused whitespace

File: snippets/focus-within.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#### Browser support
+
+<span class="snippet__support-note">⚠️ Not supported in IE11 or the current version of Edge.</span>
+
+<!-- Whenever possible, link a `caniuse` feature which allows the browser support percentage to be displayed.
+If no link is provided, it defaults to 99+%. -->
+
+- https://caniuse.com/#feat=css-focus-within
+
+<!-- tags: (separate each by a comma) -->
```
Comment (flxwu) [inline snippets/focus-within.md]: you'll need to put in some tags :)

File: snippets/focus-within.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+form:focus-within {
+  background: #f7b731;
+  color: #000000;
+}
+
+```
+
+#### Demo
+
+<!-- Leave this blank, the build script will generate the demo for you. -->
```
Comment (flxwu) [inline snippets/focus-within.md]: Please remove the template comments :)

File: snippets/focus-within.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+#### Explanation
+
+<!-- Use a step-by-step (ordered) list if possible. Keep it concise. -->
```
Comment (flxwu) [inline snippets/focus-within.md]: s. above

File: snippets/focus-within.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+#### Browser support
+
+<span class="snippet__support-note">⚠️ Not supported in IE11 or the current version of Edge.</span>
+
+<!-- Whenever possible, link a `caniuse` feature which allows the browser support percentage to be displayed.
```
Comment (flxwu) [inline snippets/focus-within.md]: s. above

**Comment by flxwu** [comment]
Yeah, the description got updated :) @chrisdemars
